statement.go

```diff
@@ -137,7 +137,7 @@ func (s *stmt) Query(args []driver.Value) (driver.Rows, error) {
 	}
 	s.rows = true
 
-	return &rows{res: &res, s: s}, nil
+	return NewRowsWithStmt(res, s), nil
```

Comment:
Is it safe to pass `duckdb_result` by value instead of reference?

Reply:
I think so, as the C API returns us the value, and even if we do it the old way we're only making a reference in `go`. I changed it just to avoid an unnecessary allocation. Also if you look at the definition of result, it's basically just a pointer within anyway.

```
type _Ctype_struct___0 struct {
	__deprecated_column_count	_Ctype_idx_t
	__deprecated_row_count		_Ctype_idx_t
	__deprecated_rows_changed	_Ctype_idx_t
	__deprecated_columns		*_Ctype_struct___1
	__deprecated_error_message	*_Ctype_char
	internal_data			unsafe.Pointer
}
```